From f1e497329c21caed9c50f051ff254af5019acef9 Mon Sep 17 00:00:00 2001
From: Vandana Kannan <vandana268@gmail.com>
Date: Thu, 31 Jan 2019 15:53:31 -0800
Subject: [PATCH 05/11] Support multi-batch input

---
 src/operator/nn/upsampling-inl.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/operator/nn/upsampling-inl.h b/src/operator/nn/upsampling-inl.h
index 7c58523e8..607981f7a 100644
--- a/src/operator/nn/upsampling-inl.h
+++ b/src/operator/nn/upsampling-inl.h
@@ -120,7 +120,6 @@ void SpatialUpSamplingNearestUpdateOutput(mshadow::Stream<cpu> *s,
   int iout[4];  // Output indices
   int iin[4];  // Input indices
 
-  channels = nbatch * channels;
   for (i0 = 0; i0 < osz0; i0++) {
     iout[0] = i0;
     iin[0] = i0;
@@ -138,8 +137,8 @@ void SpatialUpSamplingNearestUpdateOutput(mshadow::Stream<cpu> *s,
           iin[xDim] = iout[xDim] / dW;
           iin[yDim] = iout[yDim] / dH;
 
-          idst = /*i0*otensor.stride_ + i1*otensor.stride_ +*/ i2;//*otensor.stride_;
-          isrc = /*iin[0]*itensor.stride_ + iin[1]*itensor.stride_ +*/ iin[2];//*itensor.stride_;
+          idst = i0*(channels*outputHeight*outputWidth) + i1*(outputHeight*outputWidth) + i2;//*otensor.stride_;
+          isrc = iin[0]*(channels*inputHeight*inputWidth) + iin[1]*(inputHeight*inputWidth) + iin[2];//*itensor.stride_;
           if (idim > 3) {
             idst += i3*otensor.stride_;
             isrc += iin[3]*itensor.stride_;
-- 
2.20.1 (Apple Git-117)

